backend/dataall/modules/dataset_sharing/services/share_processors/s3_access_point_process_share.py

```diff
@@ -129,6 +129,7 @@ def process_revoked_shares(
         target_environment: Environment,
         source_env_group: EnvironmentGroup,
         env_group: EnvironmentGroup,
+        principal_exist: bool,
```

Comment:
Instead of passing the `principal_exist` variable I would check directly where it is used, so internally in `revoke_target_role_access_policy`

Reply:
Checking `revoke_target_role_access_policy` I realize that even if the role does not exist, the policy will exist and we need to clean it up

Reply:
Well, surprisingly, if I delete role via console, all policies are also deleted with it.
I am not sure, that we will be able to delete this role and keep policies

Reply:
Anyway, it won't hurt to check

Reply:
adressed in [b60f12d](https://github.com/data-dot-all/dataall/pull/1161/commits/b60f12d84c0cef198310dc6b022753d925816b7c)